### PR TITLE
Meed-288: Add extension for activity favorite icon

### PR DESCRIPTION
**Pull request by SaraBoutej**
In the new favorite drawer, each activity will have a new image or icon associated with its type. This PR load the activity icon associated with the activity type.

### CODE REVIEW

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
   <v-list-item class="clickable" :href="activityUrl">
     <v-list-item-icon class="me-3 my-auto">
-      <v-icon size="22" class="icon-default-color"> fas fa-file-alt </v-icon>
+      <v-icon size="24" :class="activityIcon"></v-icon>
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]:
```suggestion
      <v-icon size="24" :class="activityIcon" />
```

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-    <v-list-item-icon class="me-3 my-auto">
-      <v-icon size="22" class="icon-default-color"> fas fa-file-alt </v-icon>
+    <v-list-item-icon v-if="activityTypeExtension && activityTypeExtension.img" class="me-3 my-auto">
+      <v-img
+        :src="activityTypeExtension.img"
+        max-height="28"
+        max-width="25" />
+    </v-list-item-icon>
+    <v-list-item-icon v-else class="me-3 my-auto">
+      <v-icon size="24" :class="activityTypeExtension.class"> {{ activityTypeExtension.icon }} </v-icon>
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]:
```suggestion
    <v-list-item-icon class="me-3 my-auto">
      <v-img
        v-if="activityTypeExtension && activityTypeExtension.img"
        :src="activityTypeExtension.img"
        max-height="28"
        max-width="25" />
      <v-icon
        v-else-if="activityTypeExtension && activityTypeExtension.icon"
        v-text="activityTypeExtension.icon"
        size="24"
        :class="activityTypeExtension.class" /> 
      <v-icon
        v-else
        size="24"
        class="primary--text">
        fas fa-feather-alt
      </v-icon>
    </v-list-item-icon>
```

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+    <v-list-item-icon v-else class="me-3 my-auto">
+      <v-icon size="24" :class="activityTypeExtension.class"> {{ activityTypeExtension.icon }} </v-icon> 
     </v-list-item-icon>
 
     <v-list-item-content>
       <v-list-item-title class="text-color body-2">
         <p
           class="ma-auto"
-          v-sanitized-html="activityTitle"></p>
+          v-sanitized-html="activityTypeExtension.title"></p>
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]:
```suggestion
          v-sanitized-html="activityTitle"></p>
```

and add this computed value:

```javascript
activityTitle() {
  return this.activityTypeExtension?.title || this.activity.title;
}
```

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue [hunk 4]
```diff
@@ -32,13 +38,38 @@ export default {
...
+          }
+        });
+      } else {
+        this.activityTypeExtension.icon = this.defaultIcon;
+        this.activityTypeExtension.class = this.defaultClass;
+        this.activityTypeExtension.title = this.activity?.title ? this.favoriteTitle(this.activity.title) : this.$t('UITopBarFavoritesPortlet.label.activity');
+      }
+    }
+  },
   computed: {
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]: Delete watch and add it as computed : 

```suggestion
  }),
  computed: {
    activityTypeExtension() {
      if (this.activity?.type) {
        return this.activityExtensions.find(extension => extension.type === this.activity.type);
      }
      return null;
    },
    activityTitle() {
      return this.activityTypeExtension?.title && this.activityTypeExtension?.title(this.activity) || this.favoriteTitle(this.activity.title) || this.$t('UITopBarFavoritesPortlet.label.activity');
    }
  },
```

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue [hunk 5]
```diff
@@ -49,18 +80,12 @@ export default {
...
-          this.activityTitle = this.$t('UITopBarFavoritesPortlet.label.activity');
-        }
       });
   },
   methods: {
     favoriteTitle(title) {
       const regex = /(<([^>]+)>)/ig;
-      const activityTitle = title.replace(regex, '').split(' ').slice(0, 5).join(' ');
-      return activityTitle;
+      return title.replace(regex, '').split(' ').slice(0, 5).join(' ');
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]:
```suggestion
      return title && title.replace(regex, '').split(' ').slice(0, 5).join(' ') || '';
```

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/TopBarFavoritesDrawer.vue [hunk 6]
```diff
@@ -77,10 +81,13 @@ export default {
   },
   created() {
     this.$root.$on('open-favorite-drawer', () => {
+      document.addEventListener(`extension-${this.extensionApp}-${this.activityIconExtension}-updated`, this.refreshActivityIcon);
+      this.refreshActivityIcon();
       this.openDrawer();
     });
     this.$root.$on('close-favorite-drawer', () => {
       this.$refs.favoritesDrawer.close();
+      document.removeEventListener(`extension-${this.extensionApp}-${this.activityIconExtension}-updated`, this.refreshActivityIcon);
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/TopBarFavoritesDrawer.vue]: you should add this in `beforeDestroy` only

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue (hunk 2 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue (hunk 3 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue (hunk 4 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue (hunk 5 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/TopBarFavoritesDrawer.vue (hunk 6 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/TopBarFavoritesDrawer.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue (hunk 1 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue]: done

**Comment by boubaker** [comment]
@SaraBoutej can you rebase the PR's branch please

**Comment by SaraBoutej** [comment]
> @SaraBoutej can you rebase the PR's branch please

done

**Review by boubaker — APPROVED** [review]

**Pull request merged**